### PR TITLE
Added support for editing more metadata in webclient

### PR DESCRIPTION
**Pull request by ocherny**
Added support for editing title, artist, album, albumArtist metadata fields in web client.

### CODE REVIEW

**Comment by thebigmunch** [comment]
A couple notes:

1) You don't need to do ``s.get(field, None)``. dict.get will return None if the key doesn't exist in the dict.
2) The name of the parameter isn't very good nor is its description. For proper documentation and consistency, use the same as in [Mobileclient.change_song_metadata](https://github.com/simon-weber/gmusicapi/blob/develop/gmusicapi/clients/mobileclient.py#L192).
3) You should return the ids of the songs changed for consistency with the previous version of ``Webclient.change_song_metadata`` and ``Mobileclient.change_song_metadata``. See [here](https://github.com/simon-weber/gmusicapi/pull/229/commits/018fd6327c7776ea824b0303463afe7d4b8f65bb#diff-7f877e50f6a07120b39b748fe98bd110R295)
4) Your use of the accept_singleton decorator is incorrect. The type should be dict, not basestring. And, the documentation should read ``Changes the metadata of songs.`` again for consistency, but also because the accept_singleton decorator allows the method to take both a list of song dictionaries or a single song dictionary. So, saying it only changes metadata for a list of songs is also incorrect.

**Comment by thebigmunch** [comment]
I also forgot, you will need to re-add any necessary tests and add the method back to the Webclient documentation. See [this previous related PR](https://github.com/simon-weber/gmusicapi/pull/229).

**Comment by ocherny** [comment]
Code cleaned up, not sure how to test for metadata changes. In my experiments, it takes time for Google backend to process the data before updated metadata becomes available when requested through the mobile API (30-60 seconds, it looks like, sometimes more). The test for existing album art upload call is still commented out, not sure why.

**Comment by thebigmunch** [comment]
> Code cleaned up, not sure how to test for metadata changes. In my experiments, it takes time for Google backend to process the data before updated metadata becomes available when requested through the mobile API (30-60 seconds, it looks like, sometimes more).

Did you see the tests and comments about them in the linked pull request? Never really used the Webclient much. I'd have to test your changes sometime to see if that's also the case for me.

> The test for existing album art upload call is still commented out, not sure why.

All the Webclient tests are commented out because of [this](https://github.com/simon-weber/gmusicapi/issues/395). But, there should still be a test for this that can be used locally to check the functionality then commented out when committed until a time when Webclient tests can again be run in CI.

File: gmusicapi/protocol/webclient.py [hunk 1]
```diff
@@ -296,14 +296,25 @@ def dynamic_data(songs, session_id=""):
...
-        song_arrays = [[s['id'], None, s['albumArtUrl']] + [None] * 36 + [[]] for s in songs]
+        song_arrays = [[s['id'],
+                        s.get('title'),
+                        s.get('albumArtUrl'),
+                        s.get('artist'),
+                        s.get('album'),
+                        s.get('albumArtist')]
+                       + [None] * 33 + [[]] for s in songs]
         jsarray.append([song_arrays])
+        print(str(jsarray))
```
Comment (simon-weber) [inline gmusicapi/protocol/webclient.py]: let's also remove this extra print call

**Comment by simon-weber** [comment]
I'm fine with merging this as-is. It looks reasonable enough, and there's no real expectation for the webclient to work anyways (though we should make that clearer to users).

Thanks!

**Pull request merged**